zookeeper-server/src/main/java/org/apache/zookeeper/server/quorum/QuorumPeer.java

```diff
@@ -755,39 +769,55 @@ public void recreateSocketAddresses(long id) {
         }
     }
 
-    public InetSocketAddress getQuorumAddress(){
-        synchronized (QV_LOCK) {
-            return myQuorumAddr;
+    InetSocketAddress getQuorumAddress(){
```

Comment:
You might want to put the double-check-locking mechanism into a private getter of `myAddrs` to consolidate the logic in these 2 getters.

Reply:
Good suggestion.  Done.